ocaml-lsp-server/src/code_actions/action_mark_unused.ml

```diff
@@ -0,0 +1,26 @@
+open Import
+
+let action_title = "Mark as unused"
+
+let logfile = open_out "/tmp/ocaml-lsp.log"
+
+let log msg =
+  output_string logfile msg;
+  flush logfile
+
+let code_action_mark_value_unused doc (diagnostic : Diagnostic.t) =
+  let start = diagnostic.range.start in
+  let edit = Document.edit doc (Range.create ~start ~end_:start) "_" in
+  CodeAction.create ~diagnostics:[ diagnostic ] ~title:action_title
+    ~kind:CodeActionKind.QuickFix ~edit ~isPreferred:false ()
+
+let code_action doc (params : CodeActionParams.t) =
+  List.filter params.context.diagnostics ~f:(fun (d : Diagnostic.t) ->
+      log @@ sprintf "Diag: %s\n" d.message;
+
+      match Position.compare_inclusion params.range.start d.range with
+      | `Outside _ -> false
+      | `Inside -> true)
+  |> List.find ~f:(fun d -> Diagnostic_util.(is_unused_variable d))
```

Comment:
1. When we `List.filter` and only take one of the filtered values, we do unnecessary work. We could only `List.find`

```ocaml
List.find params.context.diagnostics ~f:(fun (d : Diagnostic.t) ->
     match Position.compare_inclusion params.range.start d.range with
      | `Outside _ -> false
      | `Inside -> Diagnostic_util.is_unused_variable d)
```

The following `Option.map` could also be inlined without much loss of readability imho. 

2. We don't usually do `open`s such as `D_util.(is_unused ...)` -- we `open` only for infix & let operators usually